main/src/ca/uwaterloo/flix/api/lsp/provider/CompleteProvider.scala

```diff
@@ -54,8 +116,34 @@ object CompleteProvider {
     // Declaration-based:
 
     // Expressed-based:
-    CompletionItem("match", "match ${1:exp} {\n case ${2:pat} => ${3:exp}\n}", None, None, CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
-    CompletionItem("query", "query ${1:db} select ${2:cols} from ${3:preds} ${4:where ${5:cond}}", None, None, CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("def", "def ${1:name}(${2:arg}:${3:type}): ${4:return_type} = \n", None, Some("code snippet to define a function"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
```

Comment:
Rename "return_type" to "type".

Reply:
I think we should have two defs one for pure functions and one for impure. The impure would look like this:

```
"def ${1:name}(${2:arg}:${3:type}): ${4:type} & Impure = \n"
```

Reply:
I am not sure if we should have the newline here or not. What do you think? If there is a newline should the next line also be indented or not?

Reply:
Indentation doesn't affect the flix program. i.e. the program works fine even if we do not keep the code properly indented. 
To keep the code properly indented so that it looks good, we can either remove `\n` or replace it with `\n\t`.

Reply:
Yes, indentation does not matter, but I want to make sure the cursor is placed in the right place.. You know for a good user experience :)

Reply:
Then, I think `\n\t` would be a better choice. What do you think?

Reply:
Agree. Lets try it.